runtime/doc/options.txt

```diff
@@ -5203,9 +5203,8 @@ A jump table for the options with a short description can be found at |Q_op|.
 	%	When included, save and restore the buffer list.  If Vim is
```

Comment:
Testing `:set shada+=%`, it looks like it is _already_ the case that `'nobuflisted'` buffers are _not_ saved/restored (check with `:ls!`). So the change in this PR actually just fixes a bug, maybe? 

Anyway, let's update the doc here:
```suggestion
	%	Save and restore the buffer list (only 'buflisted' buffers).  If Vim is
```